sources/dataset.py

```diff
@@ -261,52 +304,13 @@ def populate(self, output_dir, input_dir, nb_images=None, aggregate=False, label
             labels = p.starmap(self._preprocess, [(x, output_dir, aggregate, labelling) for x in image_list_longname])
         self.image_info = {k: v for k,v in enumerate(labels)}
 
-    def save(self, filename):
-        """Save dataset in a json file indicated by `filename`
-
-        Parameter
-        ---------
-        filename: object
-            String designing the relative path where the dataset must be saved
-        """
-        with open(filename, 'w') as fp:
-            json.dump({"image_size": self.image_size,
-                       "classes": self.class_info,
-                       "images": self.image_info}, fp)
-        utils.logger.info("The dataset has been saved into {}".format(filename))
-
-    def load(self, filename, nb_images=None):
-        """Load a dataset from a json file indicated by `filename`
-
-        Parameter
-        ---------
-        filename: object
-            String designing the relative path from where the dataset must be
-        loaded
-        nb_images: integer
-            Number of images that must be loaded (if None, the whole dataset is loaded)
-        """
-        with open(filename) as fp:
-            ds = json.load(fp)
-            self.image_size = ds["image_size"]
-            self.class_info = {int(k):ds["classes"][k] for k in ds["classes"].keys()}
-            if nb_images is None:
-                self.image_info = {int(k):ds["images"][k] for k in ds["images"].keys()}
-            else:
-                self.image_info = {int(k):ds["images"][k] for k in ds["images"].keys() if int(k) < nb_images}
-            for img_id, info in self.image_info.items():
-                info['labels'] = {int(k): v for k, v in info['labels'].items()}
-        utils.logger.info("The dataset has been loaded from {}".format(filename))
-
 class ShapeDataset(Dataset):
 
     def __init__(self, image_size, nb_classes):
         """ Class constructor
         """
-        self.image_size = image_size
-        self.class_info = defaultdict()
+        Dataset.__init__(self, image_size)
```

Comment:
It's a better practice to use the `super` function when you write inheritance in Python.
Good to read: https://docs.python.org/3/library/functions.html#super

Reply:
Fixed with `super().__init__(image_size)`